lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -77,7 +77,14 @@ Function Chocolatey-Install-Upgrade
             Fail-Json $result "Chocolatey bootstrap installation failed."
         }
         $result.changed = $true
-        $script:executable = "C:\ProgramData\chocolatey\bin\choco.exe"
+        if (Test-Path "Env:\ChocolateyInstall")
```

Comment:
So a long hunt might look like the following:
```
        $command = Get-Command -Name "choco.exe"
        if ($command)
        {
            $path = $command.Path
        }
        else
        {
            $env_value = $env:ChocolateyInstall
            if ($env_value)
            {
                $path = $env_value
            }
            else
            {
                $path = "$env:ALLUSERSPROFILE\Chocolatey\bin\choco.exe"
                if (-not (Test-Path -Path $path))
                {
                    $program_data_env = $env:ProgramData
                    if ($program_data_env)
                    {
                        $path = "$program_data_env\Chocolatey\bin\choco.exe"
                    }
                    else
                    {
                        $path = "$env:SYSTEMDRIVE\ProgramData\Chocolatey\bin\choco.exe"
                    }
                }
            }
        }

        if (-not $path -or (-not (Test-Path -Path $path)))
        {
            Fail-Json -obj $result -message "failed to find choco.exe, make sure it is added to the PATH or the env var ChocolateyInstall is set"
        }

        Add-Warning $result 'Chocolatey was missing from this system, so it was installed during this task run.'
        $script:executable = $path
```

But I think it could be shortened a bit to just:
```
$command = Get-Command -Name "choco.exe"
        if ($command)
        {
            $path = $command.Path
        }
        else
        {
            $env_value = $env:ChocolateyInstall
            if ($env_value)
            {
                $path = $env_value
            }
            else
            {
                $path = "$env:SYSTEMDRIVE\ProgramData\Chocolatey\bin\choco.exe"
            }
        }

        if (-not (Test-Path -Path $path))
        {
            Fail-Json -obj $result -message "failed to find choco.exe, make sure it is added to the PATH or the env var ChocolateyInstall is set"
        }

        Add-Warning $result 'Chocolatey was missing from this system, so it was installed during this task run.'
        $script:executable = $path
```